packages/stellar-axelar-std-derive/src/storage.rs

```diff
@@ -9,9 +12,122 @@ enum Value {
     Type(Type),
 }
 
-struct StorageAttributes {
-    storage_type: StorageType,
-    value: Value,
+impl TryFrom<&[Attribute]> for Value {
+    type Error = String;
+
+    fn try_from(attrs: &[Attribute]) -> Result<Self, Self::Error> {
+        let attr = attrs
+            .iter()
+            .filter(|attr| attr.path().is_ident("status") || attr.path().is_ident("value"))
+            .exactly_one()
+            .map_err(|_| {
+                "exactly one of #[status] and #[value(Type)] must be provided".to_string()
+            })?;
+
+        if let Meta::List(list) = &attr.meta {
+            if attr.path().is_ident("value") {
+                Ok(Self::Type(
+                    list.parse_args::<Type>()
+                        .map_err(|_| "failed to parse value type".to_string())?,
+                ))
+            } else {
+                Err("status attribute cannot have parameters".to_string())
```

Comment:
```suggestion
                Err("status attribute cannot have parameters".into())
```
auto-convert instead?